server/bleep/src/state.rs

```diff
@@ -515,9 +518,14 @@ impl StateSource {
     }
 }
 
-pub fn pretty_write_file<T: Serialize + ?Sized>(path: &Path, val: &T) -> Result<(), RepoError> {
-    let file = std::fs::File::create(path)?;
+pub fn pretty_write_file<T: Serialize + ?Sized>(
+    path: impl AsRef<Path>,
+    val: &T,
+) -> Result<(), RepoError> {
+    let tmpfile = path.as_ref().with_extension("new");
+    let file = std::fs::File::create(&tmpfile)?;
```

Comment:
We should use `File::options().create_new().open(tmpfile)` instead to avoid overwriting an existing file. If we want to be really fancy, we could use a counter or random extension with a retry mechanism.